checkov/serverless/runner.py

```diff
@@ -30,18 +30,23 @@ def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=R
 
         if files:
             for file in files:
-                if file in SLS_FILE_MASK:
+                if os.path.basename(file) in SLS_FILE_MASK:
                     parse_result = parse(file)
                     if parse_result:
                         (definitions[file], definitions_raw[file]) = parse_result
 
         if root_folder:
             for root, d_names, f_names in os.walk(root_folder):
+                # Don't walk in to "node_modules" directories under the root folder. If –for some reason–
+                # scanning one of these is desired, it can be directly specified.
+                if "node_modules" in d_names:
+                    d_names.remove("node_modules")
```

Comment:
would it work for internal dirs too?

Reply:
Do you mean nested? For example: “<root_folder>/node_modules” as well as “<root_folder>/something/node_modules”? If so, yes. It’s called at each directory with the sub-dir names that will traversed if left alone.